src/lib/fastDownward/fastDownward.ts

```diff
@@ -98,13 +98,37 @@ export function search<LU extends LearningUnit>(
 	const pathList: Path[] = [];
 	const openListExtra = [];
 	let duration = 0;
+	let skillsNotFound: string[] = [];
 
 	while (openList.length > 0) {
 		//openList.sort((a, b) => a.heuristic - b.heuristic); // Replaced by inserting newNode to openList in sorted manner
 		const startTime = new Date().getTime();
 
 		const currentNode = openList.shift()!;
 
+		// Get the skill in the goal but not in the currentNode status
+		const remainSkillTemp: string[] = [];
```

Comment:
Should we add remainSkillTemp to SearchNode / State?

Reply:
remainSkillTemp is related to all goal skills (used to check all SearchNode), not related to any specific SearchNode or state.
So, I would suggest keeping it out of the SearchNode / State.